zsl/interface/gearman/task_filler.py

```diff
@@ -7,32 +7,27 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-# Initialize app
-from zsl.interface.gearman import importer
-importer.append_asl_path_to_pythonpath()
-from zsl.interface.importer import initialize_cli_application, InitializationContext
-initialize_cli_application(InitializationContext(unit_test=False))
 
-import sys
 import gearman
 import json
-from zsl.application.service_application import service_application as app
 
-if __name__ == "__main__":
-    if len(sys.argv) != 3:
-        sys.stderr.write("Usage: task_filler.py <task name> <task data (file:// for file input, otherwise raw data)>\n")
-        exit(1)
+from zsl import Config, inject, Injected
 
-    print("Initializing client.")
-    gm_client = gearman.GearmanClient(["{0}:{1}".format(app.config['GEARMAN']['host'], app.config['GEARMAN']['port'])])
 
+@inject(config=Config)
+def exec_task_filler(task_path, json_data, config=Injected):
+    print("Initializing client.")
+    gm_client = gearman.GearmanClient(["{0}:{1}".format(config['GEARMAN']['host'], config['GEARMAN']['port'])])
     print("Client initialized.")
-    json_data = sys.argv[2]
+
     if json_data.startswith("file://"):
```

Comment:
This one should be done where the command line/input is parsed. The code here does a high-level part of calling the Gearman not data parsing.